api/api/views/media_views.py

```diff
@@ -87,6 +89,20 @@ def get_db_results(self, results):
 
     # Standard actions
 
+    def retrieve(self, request, *_, **__):
+        @dataclass
+        class PseudoHit:  # implements Identifiable protocol
+            identifier: str
+
+        instance = self.get_object()
+        hit = PseudoHit(identifier=str(instance.identifier))
+
+        search_context = SearchContext.build([hit], self.default_index).asdict()
```

Comment:
FWIW, `SearchContext::build` could be simplified if it just received the list of identifiers. Considering it gets mapped in the `::build` function out of the `results` list anyway, and those results aren't used in any _other_ way, I think it would be a nice way to simplify the `::build` function and the type juggling we're doing. It prevents us from needing to create a thin container class to satisfy `SearchContext` and reduces `SearchContext`'s concerns by restricting it to just the identifiers, the part of the results it actually cares about.

```suggestion
        instance = self.get_object()
        search_context = SearchContext.build([instance], self.default_index).asdict()
```